python/triton/experimental/gluon/language/_layouts.py

```diff
@@ -340,59 +347,52 @@ class NVMMASharedLayout(SharedLayout):
         rank (int): Rank of the tensor.
         transposed (bool): Whether the layout is transposed.
         fp4_padded (bool): Whether FP4 padding is used.
-        ctas_per_cga (Optional[List[int]]): CTAs per CGA grouping.
-        cta_split_num (Optional[List[int]]): Split factors for CTAs.
-        cta_order (Optional[List[int]]): CTA ordering.
+        cga_layout (Optional[List[List[int]]]): Bases describing CTA tiling.
     """
     swizzle_byte_width: int
     element_bitwidth: int
-    rank: int
+    rank: int = 2
```

Comment:
Not sure we want this default

Reply:
We need rank > 2 for TMA and use it regularly. So, no I don't think we should leave this in.